### PR TITLE
Integration of AWS and Azure. Signed-off-by: Sumit Pathak <pathaksu@v…

### PR DESCRIPTION
**Pull request by ajain0811**
…mware.com>

### CODE REVIEW

File: azure/src/lint.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  req.on('error', error => reject(error));
+
+  if (postData) {
+    req.write(postData);
+  }
+
+  req.end();
+});
+
+const gunzipData = zippedData => new Promise((resolve, reject) => {
```
Comment (satyapraveen2002) [inline azure/src/lint.js:76]: Can you check if this method is needed?

File: azure/src/lint.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  zlib.gunzip(
+    zippedData,
+    (error, unzippedLogs) => (error ? reject(error) : resolve(unzippedLogs))
+  );
+});
+
+const gzipLogs = logsJson => new Promise((resolve, reject) => {
```
Comment (satyapraveen2002) [inline azure/src/lint.js:83]: Same case here.

File: azure/test/lint.test.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      field4: { field41: ['a', 'b', 'c'] },
+    };
+    const result = flattenJson(exampleJson);
+    expect(result).toMatchSnapshot();
+  });
+});
+
+describe('LIntHttpEnv', () => {
+  it('should create options for the HTTP stream', () => {
+    const STRUCTURE = 'cloudwatch';
```
Comment (satyapraveen2002) [inline azure/test/lint.test.js:47]: In Azure, we are seeing test cases for AWS. Can you please check it out

File: azure/test/lint.test.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    expect(options.hostname).toBe('data.test.com');
+    expect(options.path).toBe('/le-mans/v1/streams/logiq-rawlogs');
+  });
+});
+
+describe('gzipLogs and gunzipData', () => {
+  const logsJson = {
+    logGroup: '/aws/lambda/runbook',
+    subscriptionFilters: [
+      'LambdaStream_CloudWatchLogCollector',
```
Comment (satyapraveen2002) [inline azure/test/lint.test.js:77]: same case here

**Review by satyapraveen2002 — COMMENTED** [review]
Added minor comments

File: azure/src/lint.js (hunk 1 above)
Comment (sumitpathak) [reply azure/src/lint.js]: Removed unused method

File: azure/src/lint.js (hunk 2 above)
Comment (sumitpathak) [reply azure/src/lint.js]: Removed unused method

File: azure/test/lint.test.js (hunk 3 above)
Comment (sumitpathak) [reply azure/test/lint.test.js]: Removed aws test cases

File: azure/test/lint.test.js (hunk 4 above)
Comment (sumitpathak) [reply azure/test/lint.test.js]: Removed aws test cases

File: azure/target/blobstorage/blobStorageFunction/index.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
+/*
```
Comment (satyapraveen2002) [inline azure/target/blobstorage/blobStorageFunction/index.js:1]: Please check if any code related to Eventhub exists, if so please remove it.

File: azure/target/blobstorage/blobStorageFunction/index.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+/**
+ * Converts raw json string to json object for blob.
+ * @param logText
+ * @returns {{}}
+ */
+const processLogTextAsJson = (logText) => {
+    const keys = [];
+    const mergedRecords = {};
+    //mergedRecords.blob = logText;
```
Comment (slk244) [inline azure/target/blobstorage/blobStorageFunction/index.js:34]: remove this

File: azure/target/blobstorage/blobStorageFunction/index.js [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+                record.log_type = 'azure_log';
+                record.logsource = logSource;
+            });
+        }
+    }
+};
+
+/**
+ * Fetch service name and provider from the resource ID string
+ * and adding those properties to event_provider,eventsource.
```
Comment (slk244) [inline azure/target/blobstorage/blobStorageFunction/index.js:85]: please add an example of how resourceID will look like and the end result on what this method. does with that

File: azure/target/blobstorage/blobStorageFunction/index.js [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+            .trim()
+            .split('\n');
+    } else {
+        logs = JSON.stringify(serviceLogs)
+            .trim()
+            .split('\n');
+    }
+    logs.forEach(log => {
+        const data = collector.processLogsJsonFromBlobStorage(log);
+        return collector.postDataToStream(data);
```
Comment (slk244) [inline azure/target/blobstorage/blobStorageFunction/index.js:142]: Here wont it return as soon as the first log gets processed?
i meant further logs will never be sent to LINT

File: azure/target/blobstorage/blobStorageFunction/index.js [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+}
+
+/**
+ * Error handeling.
+ * @param error
+ * @param context
+ */
+const handleError = (error, context) => {
+    context.fail(error);
+    console.log(error);
```
Comment (slk244) [inline azure/target/blobstorage/blobStorageFunction/index.js:180]: just cross check if the log should be above context.fail()

log first and then fail

**Review by slk244 — COMMENTED** [review]
few comments

File: azure/target/blobstorage/blobStorageFunction/index.js (hunk 6 above)
Comment (sumitpathak) [reply azure/target/blobstorage/blobStorageFunction/index.js:34]: Removed commented statement

File: azure/target/blobstorage/blobStorageFunction/index.js (hunk 7 above)
Comment (sumitpathak) [reply azure/target/blobstorage/blobStorageFunction/index.js:84]: Added example resourceId and result

File: azure/target/blobstorage/blobStorageFunction/index.js (hunk 8 above)
Comment (sumitpathak) [reply azure/target/blobstorage/blobStorageFunction/index.js]: Have removed return statement and tested functionality, working as expected.

File: azure/target/blobstorage/blobStorageFunction/index.js (hunk 9 above)
Comment (sumitpathak) [reply azure/target/blobstorage/blobStorageFunction/index.js:181]: Implemented review comments

**Review by slk244 — APPROVED** [review]

**Pull request merged**